### PR TITLE
Blacklist specific plugins within an extension package by ID.

### PR DESCRIPTION
**Pull request by afshin**
cc: @ivanov

### CODE REVIEW

File: jupyterlab/index.js [hunk 1]
```diff
@@ -5,21 +5,30 @@ __webpack_public_path__ = PageConfig.getOption('publicUrl');
...
 // This needs to come after __webpack_public_path__ is set.
 require('font-awesome/css/font-awesome.min.css');
-var app = require('@jupyterlab/application').JupyterLab;
+var App = require('@jupyterlab/application').JupyterLab;
 
 
 function main() {
-    var version = PageConfig.getOption('appVersion') || 'unknown';
+    var assetsDir = PageConfig.getOption('assetsDir') || '';
+    var blacklist = PageConfig.getOption('blacklist') || [];
```
Comment (blink1073) [inline jupyterlab/index.js]: This is redundant with the usage below.

File: jupyterlab/index.js [hunk 2]
```diff
@@ -5,21 +5,30 @@ __webpack_public_path__ = PageConfig.getOption('publicUrl');
...
+    var version = PageConfig.getOption('appVersion') || 'unknown';
 
     if (version[0] === 'v') {
         version = version.slice(1);
     }
 
+    var blacklist = [];
+    try {
+        var option = PageConfig.getOption('blacklist');
+        disabled = JSON.parse(option);
```
Comment (blink1073) [inline jupyterlab/index.js]: should be `blacklist`

**Comment by ivanov** [comment]
Thanks for looping me in. What's the distinction between having a plugin in "ignored" vs "blacklisted"?  Do blacklisted plugins not get loaded via the dependency resolution mechanism even when there's something that depends on them? It would be good to capture this in the documentation of these two options, since they seem to be related, but (maybe?) have slightly different semantics.

**Comment by blink1073** [comment]
@afshin and I talked about this on Friday, and agree the terms are a bit confusing.  We propose the following moving forward:

- `disabledExtensions` will be called `disabledModules` so that "extension" and "plugin" can remain synonyms, and a "module" is a superset of extension
- `ignoredPlugins` will be called `deferredPlugins` because only their `autostart` is being ignored, but they are actually available at runtime
- `disabledPlugins` will be the individual plugins that are blacklisted from being loaded when modules load

**Pull request closed without merging**